packages/stores/src/ui-config/convert-to-stake.ts

```diff
@@ -0,0 +1,264 @@
+  poolId: string;
+  totalValue: PricePretty;
+  userPoolAssets: CoinPretty[];
+  currentApr: RatePretty;
+};
+
+/** Aggregates user balancer shares are available to be staked, preferably for a higher APR.
+ *  Manages the state for the selection process, with selections keyed by pool ID. */
+export class UserConvertToStakeConfig {
+  @observable
+  protected _selectedConversionPoolIds = new Set<string>();
+
+  @computed
+  get selectedConversionPoolIds() {
+    return this._selectedConversionPoolIds;
+  }
+
+  @computed
+  get selectedConversions() {
+    return this.suggestedConvertibleAssetsPerPool.filter(({ poolId }) =>
+      this.selectedConversionPoolIds.has(poolId)
+    );
+  }
+
+  /** For each user owned pool, this provides a breakdown of shares
+   *  that are of lesser returns than the staking inflationary returns. */
+  @computed
+  get suggestedConvertibleAssetsPerPool(): SuggestedConvertToStakeAssets[] {
+    const ownedSharePoolIds = this.osmosisQueries.queryGammPoolShare
+      .getOwnPools(this.accountAddress)
+      .filter((poolId) => {
+        // only share pools
+        const queryPool = this.osmosisQueries.queryPools.getPool(poolId);
+        if (Boolean(queryPool?.sharePool)) return true;
+        return false;
+      });
+
+    const account = this.osmosisAccount;
+    const stakeCurrency = this.chainGetter.getChain(
+      this.osmosisChainId
+    ).stakeCurrency;
+
+    if (!account) return [];
+
+    const conversions: SuggestedConvertToStakeAssets[] = [];
+
+    // calculate conversions for user's share pools
+    ownedSharePoolIds.forEach((sharePoolId) => {
+      const { sharePoolDetail, superfluidPoolDetail, poolBonding } =
+        this.derivedDataStore.getForPool(sharePoolId);
+
+      // only include if contains stake currency
+      if (
+        !sharePoolDetail.querySharePool?.hasPoolAsset(
+          stakeCurrency.coinMinimalDenom
+        )
+      )
+        return;
+
+      // only include pools with > 1¢ value
+      // TODO: temporarily only look at bonded value, since we can't serialize this case in the message
+      if (
+        !(
+          sharePoolDetail.userStats?.bondedValue?.toDec().isPositive() ||
+          sharePoolDetail.userUnlockingAssets.length > 0
+        )
+      )
+        return;
+
+      const totalValue =
+        sharePoolDetail.userStats?.totalShareValue ??
+        new PricePretty(this.fiatCurrency, 0);
+      const userPoolAssets = sharePoolDetail.userPoolAssets.map(
+        ({ asset }) => asset
+      );
+
+      const currentApr =
+        poolBonding.highestBondDuration?.aggregateApr ??
+        superfluidPoolDetail.superfluidApr.add(sharePoolDetail.swapFeeApr);
+
+      // only include if better opportunity
+      if (currentApr.toDec().lte(this.stakeApr.toDec().quo(new Dec(100)))) {
+        conversions.push({
+          poolId: sharePoolId,
+          totalValue,
+          userPoolAssets,
+          currentApr,
+        });
+      }
+    });
```

Comment:
hat do you think about using the reduce function? I believe using it would simplify understanding the loop

```tsx
const conversions: SuggestedConvertToStakeAssets[] = ownedSharePoolIds.reduce(
  (acc, sharePoolId) => {
    const { sharePoolDetail, superfluidPoolDetail, poolBonding } =
      this.derivedDataStore.getForPool(sharePoolId);

    if (
      !sharePoolDetail.querySharePool?.hasPoolAsset(
        stakeCurrency.coinMinimalDenom
      )
    ) {
      return acc;
    }

    if (
      !(
        sharePoolDetail.userStats?.bondedValue?.toDec().isPositive() ||
        sharePoolDetail.userUnlockingAssets.length > 0
      )
    ) {
      return acc;
    }

    const totalValue =
      sharePoolDetail.userStats?.totalShareValue ??
      new PricePretty(this.fiatCurrency, 0);
    const userPoolAssets = sharePoolDetail.userPoolAssets.map(
      ({ asset }) => asset
    );

    const currentApr =
      poolBonding.highestBondDuration?.aggregateApr ??
      superfluidPoolDetail.superfluidApr.add(sharePoolDetail.swapFeeApr);

    if (currentApr.toDec().lte(this.stakeApr.toDec().quo(new Dec(100)))) {
      acc.push({
        poolId: sharePoolId,
        totalValue,
        userPoolAssets,
        currentApr,
      });
    }

    return acc;
  },
  []
);

```